bull-transformer-generator/transformer-generator-core/src/main/java/com/hotels/beans/generator/core/mapping/JavaBeanCode.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beans.generator.core.mapping;
+
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+import com.squareup.javapoet.CodeBlock;
+
+/**
+ * A specialization of {@link MappingCode} that generates code for mapping between two JavaBeans.
+ * The creation code uses the default no-arg constructor.
+ * The initialization code uses the available destination setters to map the source getters.
+ * @author mmirk
+ */
+class JavaBeanCode extends MappingCode {
+    /**
+     * The source type.
+     */
+    private final Class<?> source;
+    /**
+     * The destination type.
+     */
+    private final Class<?> destination;
+
+    /**
+     * Instantiates a new JavaBean code model, source and destination types must be JavaBeans.
+     * @param sourceBean the source
+     * @param destinationBean the destination
+     */
+    JavaBeanCode(final Class<?> sourceBean, final Class<?> destinationBean) {
+        this.source = sourceBean;
+        this.destination = destinationBean;
+    }
+
+    @Override
+    protected CodeBlock creation() {
```

Comment:
please add a JavaDoc for this method

Reply:
Usually Javadoc on overrides is not needed since it's inherited from parent doc. Unless particular changes in the overridden implementation that need to be documented, I think this and other subclasses adhere to the general contract already documented.
Is this mandated by the project style?

Reply:
ok so please add:
```
/**
 * {@inheritDoc}
 */
```